### PR TITLE
 PLP-Pipeline Blog Series Part 2: From Raw Clinical Data to Predictive Models

### PR DESCRIPTION
**Pull request by kosuri-indu**

### CODE REVIEW

File: posts/indu-plp-part2/plp-part2.qmd [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+engine: julia
+image: false
+categories:
+  - patient-level prediction
+  - omop cdm
+  - observational health
+  - machine learning
+---
+
+# Introduction 👋
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:18]: I think you should also mention at the end of this section the steps you are going to take here or that you are guiding the reader through a prediction pipeline.

File: posts/indu-plp-part2/plp-part2.qmd [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+In [Part 1](../indu-plp-part1plp-part1.qmd), we formulated a research question, loaded synthetic patient data, and constructed our target (hypertension) and outcome (diabetes) cohorts using OHDSI definitions and Julia tools. If you haven’t read that yet, I recommend checking it out before diving in here.
+
+Now in Part 2, we move from cohorts to models - i.e., we’ll bridge the gap between structured clinical data and predictive modeling. Specifically, we’ll walk through:
+
+- Extracting patient-level features using cohort definitions
+- Performing preprocessing (imputation, encoding, normalization)
+- Splitting data into train/test sets
+- Training ML models using MLJ.jl
+
+# Reference: Foundation from the OHDSI PLP Framework
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:31]: You can cut this section on the Reference paper

File: posts/indu-plp-part2/plp-part2.qmd [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+We extract features from several structured OMOP CDM tables such as:
+
+- `condition_occurrence`
+- `drug_exposure`
+- `procedure_occurrence`
+- `observation`
+- `measurement`
+- `person`
+
+These features are computed for each patient in the **target cohort** using a **365-day window prior to the cohort start date**, aligning with the standardized methodology described in the OHDSI PLP framework by Reps et al. (JAMIA 2018).
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:57]: What features are you extracting from these tables?

File: posts/indu-plp-part2/plp-part2.qmd [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+Each domain is queried independently, and the results are merged on `subject_id` to create a patient-level feature matrix for modeling.
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:59]: These are not domains but rather tables -- could you clarify that?

File: posts/indu-plp-part2/plp-part2.qmd [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+# Limit events to the 365 days before the cohort start date
+AND de.drug_exposure_start_date BETWEEN c.cohort_start_date - INTERVAL 365 DAY AND c.cohort_start_date
+
+# Aggregate by subject
+GROUP BY c.subject_id
+"""
+```
+
+This approach is repeated across other domains (conditions, observations, etc.) using similar logic-adhering to the paper’s recommendation to generate temporal, interpretable, and structured patient features from OMOP CDM.
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:111]: These are not domains but rather tables -- could you clarify that?

File: posts/indu-plp-part2/plp-part2.qmd [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+
+# Step 2: Attaching Outcome Labels
+
+Now that we have extracted features for the target population (patients diagnosed with hypertension), the next step is to attach outcome labels that indicate whether a patient later developed diabetes.
+
+We use **Cohort 2**, which contains patients from the hypertension cohort who were subsequently diagnosed with diabetes, as defined using `OHDSICohortExpressions.jl`. This ensures that diabetes occurs *after* the hypertension event, maintaining proper temporal ordering between target and outcome.
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:118]: Could you specify if this is the outcome or target cohort?

File: posts/indu-plp-part2/plp-part2.qmd [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+This gives us a binary classification dataset with:
+
+- $1$: patient developed diabetes after hypertension
+- $0$: patient did not develop diabetes (or not within the observed period)
+
+This labeled dataset is now ready for preprocessing and model training in later steps.
+
+<br>
+<center>
+  ![](./binary_classification.png)
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:150]: I feel like this diagram here is a little out of place -- should you put this diagram closer to the ML part?

File: posts/indu-plp-part2/plp-part2.qmd [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+  Binary Classification Model
+</center>
+
+# Step 3: Preprocessing for Modeling
+
+Once we have the raw feature matrix and outcome labels, the next crucial step is to preprocess the dataset to make it suitable for training machine learning models. This step is essential because raw healthcare data, even when structured via the OMOP CDM, often contains missing values, variable scales, and unencoded categorical information. These issues can affect model performance, stability, and interpretability.
+
+The preprocessing approach we use is directly inspired by best practices discussed in the JAMIA paper. The authors highlight the importance of handling missing values, standardizing predictors, and transforming categorical data into usable formats.
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:159]: You can cut this paragraph

File: posts/indu-plp-part2/plp-part2.qmd [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+train, test = partition(eachindex(df.outcome), 0.8, shuffle=true)
+```
+<br>
+<center>
+  ![](./train_test_splitting.png)
+
+  Train-Test splitting (80% - 20%)
+</center>
+
+# Step 4: Model Training with MLJ.jl
```
Comment (TheCedarPrince) [inline posts/indu-plp-part2/plp-part2.qmd:205]: I really like the code snippets but could you make this section more concise? Thanks!

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Great draft here @kosuri-indu ! I left some fine tuning comments but otherwise, this looks really great! :D

File: posts/indu-plp-part2/plp-part2.qmd (hunk 7 above)
Comment (kosuri-indu) [reply posts/indu-plp-part2/plp-part2.qmd:150]: I placed the diagram there because it was related with the 1 and 0 for diabetes mentioned just above it. Now that you are mentioned, I think it would fit better near the MLJ section. I will go ahead and move it there. 👍

**Comment by kosuri-indu** [comment]
Hey @TheCedarPrince, I have made the changes you mentioned. Could you please review them and let me know if everything looks good?

**Comment by TheCedarPrince** [comment]
Before I review again @kosuri-indu, could you resolve the merge conflicts here?

**Comment by kosuri-indu** [comment]
@TheCedarPrince Resolved merge conflicts here as well 😄

File: docs/search.json
Comment (TheCedarPrince) [inline docs/search.json:1]: Looks like you still have some unresolved conflicts here. Could you fix this?

File: posts/indu-plp-part2/plp-part2.qmd (hunk 1 above)
Comment (TheCedarPrince) [reply posts/indu-plp-part2/plp-part2.qmd]: Nah, I think if you have done those other edits, thinking again, I think this is perfectly fine now.

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Only two last comments and then this should be ready!

**Comment by kosuri-indu** [comment]
Hey @TheCedarPrince,
Really sorry, I didn’t see those earlier. I think I’ve resolved all of them now. There were a few bootstrap version changes as well, are those fine?
